openlibrary/i18n/__init__.py

```diff
@@ -101,6 +103,34 @@ def load_translations(lang):
     if os.path.exists(mo_path):
         return Translations(open(mo_path, 'rb'))
 
+
+def get_ol_locale():
```

Comment:
To make this a little easier to test (i.e. avoid having to mock `web.ctx`), I often do something like this:

```suggestion
def get_ol_locale(locale_cookie=web.cookies().get('i18n_code'), browser_accepted_langs=web.ctx.env.get('HTTP_ACCEPT_LANGUAGE')):
```

That way the actual code can easily use this function, but in the tests, if I want to test specific flows, I can override them easily.

Reply:
Done. Using the `i18n_loadhook` now.